### ISSUE TITLE
TorGuard - No Outside Connection

### ISSUE THREAD

**Issue by Muckoma** [haugene/docker-transmission-openvpn#1150]
Hi All,
I was on a whirlpool of trying different VPN providers to see which gave me the best results. I was using NORDVPN and then realized it doesn't allow me to port forward (seeding was terrible. I'm trying to enable P2P). I switched to PIA, which seems to be working okay but slow upload speeds. (40 Mbps outside docker and 10-15 Mbps inside docker).

I am now trying Torguard but I cannot set it to ping google.com. Below is my Docker settings. I am creating the docker image from the Synology docker interface so not sure if I'm missing any settings. I did check the same credentials on my phone and it seems to connect. Not sure what I'm missing. 

PS,
I'm planning to add to the README on how people can set up port forwarding and get things up and running for various providers. It seems to be missing in the READMEs and causes a lot of confusion. 

EDIT #2 - Would a docker compose be better? 

EDIT #3 - Fixed this issue by using Docker Compose and adding DNS. Next question, how do I go about selecting server for TorGuard. I tried `OPENVPN_CONFIG=USA-MIAMI` but that was causing my container to crash.
`version: '2'
services:
 transmission:
  image: haugene/transmission-openvpn
  container_name: TorTransmission
  cap_add:
    - NET_ADMIN
  devices:
    - /dev/net/tun
  restart: always
  ports:
    - "9091:9091"
    - "8888:8888"
  dns:
    - 8.8.8.8
    - 8.8.4.4
  volumes:
    - /etc/localtime:/etc/localtime:ro
    - /volume1/data/downloads/:/data
    - /volume1/data/downloads/transmission-home/transmission-pre-start.sh:/scripts/transmission-pre-start.sh
    - /volume1/data/downloads/transmission-home/transmission-postprocess.sh:/scripts/transmission-postprocess.sh
  environment:
    - OPENVPN_PROVIDER=TORGUARD
    - OPENVPN_USERNAME=<Removed>
    - OPENVPN_PASSWORD=<Removed>
    - OPENVPN_OPTS=--inactive 3600 --ping 10 --ping-exit 60
    - LOCAL_NETWORK=192.168.0.0/24
    - PUID=1026
    - PGID=100
    - TRANSMISSION_PORT_FORWARDING_ENABLED=true
    - TRANSMISSION_RPC_HOST_WHITELIST=192.168.86.21
    - TRANSMISSION_RPC_PASSWORD=eatmyshorts
    - TRANSMISSION_RPC_PORT=9091
    - TRANSMISSION_RPC_USERNAME=abin
    - TRANSMISSION_RPC_WHITELIST= 127.0.0.1, 192.168.86.*
    - TRANSMISSION_SCRIPT_TORRENT_DONE_FILENAME=/data/transmission-home/transmission-postprocess.sh`

**Comment by pkishino** [comment]
looks like net-admin missing
--cap-add=NET_ADMIN

**Comment by pkishino** [comment]
regarding config..check what the config for the server is called

**Comment by pkishino** [comment]
what are the logs for when you select USA-MIAMI?

**Comment by Muckoma** [comment]
@pkishino - It seems like Synology NAS doesn't add --cap-add=NET_ADMIN when using the UI. Guess you have to go docker compose route. 

In regards to config:
I got the file name from here. https://github.com/haugene/docker-transmission-openvpn/tree/master/openvpn/torguard

Here is the log
`

4/21/2020 7:09 | stdout | Setting OPENVPN credentials... |  
-- | -- | -- | --
  |   |   |  
4/21/2020 7:09 | stdout | Starting OpenVPN using config USA-MIAMI.ovpn |  
  |   |   |  
4/21/2020 7:09 | stdout | Using OpenVPN provider: TORGUARD |  
  |   |   |  
4/21/2020 7:09 | stdout | Use --help for more information. |  
  |   |   |  
4/21/2020 7:09 | stdout | Options error: Parameter ca_file   can only be specified in TLS-mode, i.e. where --tls-server or --tls-client is   also specified. |  
4/21/2020 7:09 | stdout | adding route to local network 192.168.0.0/24 via 172.21.0.1 dev eth0 |  
  |   |   |  
4/21/2020 7:09 | stdout | Setting OPENVPN credentials... |  
  |   |   |  
4/21/2020 7:09 | stdout | Starting OpenVPN using config USA-MIAMI.ovpn |  
  |   |   |  
4/21/2020 7:09 | stdout | Using OpenVPN provider: TORGUARD |  
  |   |   |  

`

**Comment by pkishino** [comment]
hmm, nothing after that? seems like it found the ovpn and loading it

**Comment by Muckoma** [comment]
> hmm, nothing after that? seems like it found the ovpn and loading it

Yeah, it just seems to repeat the same message because it restarts

**Comment by pkishino** [comment]
Hmm, you could try the rebind from command line in the container from the other ticket that you commented on. 
Strange that it doesn’t show more log

**Comment by Muckoma** [comment]
@pkishino are you referring to this post & this command?  
https://github.com/haugene/docker-transmission-openvpn/issues/1135

transmission-remote -p 51413

I'm assuming I run this command inside the docker container? Just ssh in and run it?

**Comment by rbrussell82** [comment]
I'm having a similar issue but Transmission actually shows my port as open, I just can't seed anything.  Opened a ticket with Torguard and they verified all of my OpenVPN settings are correct.

**Comment by pkishino** [comment]
I’m using one of the Tokyo servers and works great, speeds are fast enough,open port is 1080.
But I’ve also tested kista server in Sweden and worked great too.
@rbrussell82 hmm, if port is open.. no seed on any torrent? Even high traffic ones?

**Comment by Muckoma** [comment]
@pkishino - Tried JP Tokyo 1 but port status shows unknown still in Transmission. 

Here are the logs.
`[2020-04-25 00:12:44.499] RPC Server Adding address to whitelist: 192.168.86.21 (rpc-server.c:971)
[2020-04-25 00:12:44.499] RPC Server Serving RPC and Web requests on port 127.0.0.1:9091/transmission/ (rpc-server.c:1213)
[2020-04-25 00:12:44.499] UDP Failed to set receive buffer: requested 4194304, got 425984 (tr-udp.c:84)
[2020-04-25 00:12:44.499] UDP Please add the line "net.core.rmem_max = 4194304" to /etc/sysctl.conf (tr-udp.c:89)
[2020-04-25 00:12:44.499] UDP Failed to set send buffer: requested 1048576, got 425984 (tr-udp.c:95)
[2020-04-25 00:12:44.499] UDP Please add the line "net.core.wmem_max = 1048576" to /etc/sysctl.conf (tr-udp.c:100)
[2020-04-25 00:12:44.499] Using settings from "/data/transmission-home" (daemon.c:528)
[2020-04-25 00:12:44.499] Saved "/data/transmission-home/settings.json" (variant.c:1266)
[2020-04-25 00:12:44.499] Watching "/data/watch" for new .torrent files (daemon.c:573)
[2020-04-25 00:12:44.499] Loaded 2 torrents (session.c:2034)
[2020-04-25 00:12:44.499] Port Forwarding (NAT-PMP) initnatpmp succeeded (0) (natpmp.c:70)
[2020-04-25 00:12:44.499] Port Forwarding (NAT-PMP) sendpublicaddressrequest succeeded (2) (natpmp.c:70)
[2020-04-25 00:12:52.508] Port Forwarding State changed from "Not forwarded" to "Starting" (port-forwarding.c:92)
[2020-04-25 00:12:52.508] Port Forwarding State changed from "Starting" to "???" (port-forwarding.c:92)
[2020-04-25 00:13:15.500] Port Forwarding Stopped (port-forwarding.c:180)
[2020-04-25 00:13:45.501] Port Forwarding (NAT-PMP) initnatpmp succeeded (0) (natpmp.c:70)
[2020-04-25 00:13:45.501] Port Forwarding (NAT-PMP) sendpublicaddressrequest succeeded (2) (natpmp.c:70)
[2020-04-25 00:13:53.500] Port Forwarding State changed from "Not forwarded" to "Starting" (port-forwarding.c:92)
[2020-04-25 00:13:53.500] Port Forwarding State changed from "Starting" to "???" (port-forwarding.c:92)`

**Comment by pkishino** [comment]
Hmm, just checked my config.. I use:
remote jp-tok.pvdata.host 1195 udp

**Comment by Muckoma** [comment]
> Hmm, just checked my config.. I use:
> remote jp-tok.pvdata.host 1195 udp

So is tht a custom one you downloaded from the website? I looked through all the VPN files in the repo and none fo them use 1194. It's either 443 or 1194.

**Comment by pkishino** [comment]
Hmm, it’s the same as standard Tokyo one but different port, I think I used
the app first to verify then just copied.. I remember asking private VPN
about it, perhaps the ovpn files need updating, let me check

**Comment by Muckoma** [comment]
Tried Kista and no luck there either. Logs show same thing as before..Transmission still showing unknown. Strange..

**Comment by pkishino** [comment]
Which port are you trying? I use 1080 and it’s working for me in kista and tokyo(udp 1195 for that one) ..I know they have dedicated torrent servers, perhaps check if they have updated the ovpn files recently or connect with the app and check the settings it uses, might need to update..

**Comment by Muckoma** [comment]
> Which port are you trying? I use 1080 and it’s working for me in kista and tokyo(udp 1195 for that one) ..I know they have dedicated torrent servers, perhaps check if they have updated the ovpn files recently or connect with the app and check the settings it uses, might need to update..

I'm trying to use 1080 like you. Here is my docker-compose file. I tried using the stockholm dedicated one where all ports are forward but even that isn't working. Any ideas? PIA works for me so not sure why PrivateVPN wouldn't work. 

[docker-compose.txt](https://github.com/haugene/docker-transmission-openvpn/files/4534637/docker-compose.txt)

**Comment by pkishino** [comment]
just tried that allforward and it says closed..might  need to check with privatevpn support about this..

**Comment by Muckoma** [comment]
> just tried that allforward and it says closed..might need to check with privatevpn support about this..

So I looked at one of the dedicated servers for Torrents (IT-Milan) in the PrivateVPN client. It says ALL ports are forwarding in it. I downloaded the latest PrivateVPN-TUN --> UPD --> *.ovpn files and the one for IT-Milan look different than the one in the repo. I'm not sure if these difference make a difference but here is what I've noticed. I also attached the latest file for your reference. I also tried SE Kista but no luck there either.

From PrivateVPN website has these in the Options:
`tun-ipv6`

Our opvn file has these extra settings in options. Not sure if these make a difference.
`pull-filter ignore "DNS"
pull-filter ignore "ping"
ping 30
ping-exit 120`
